src/App.tsx

```diff
@@ -1,14 +1,48 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loaded, setLoaded] = useState(false);
+  const [selectedOption, setSelectedOption] = useState('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | undefined>(undefined);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      const result = await getTodos();
```

Comment:
Consider adding error handling for the `getTodos` API call. This will help manage any potential errors that occur during the data fetching process.